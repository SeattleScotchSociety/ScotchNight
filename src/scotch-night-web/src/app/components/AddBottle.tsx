import { inject, observer } from "mobx-react";
import * as React from "react";
import { Button, ButtonGroup } from "reactstrap";

import { Bottle, IBottle } from "../stores/BottleStore";
import { IRootStore } from "../stores/RootStore";

import { MyNotes } from "./MyNotes";
import { NotesOverview } from "./NotesOverview";

interface IAddBottleProps { store: IRootStore; }
interface IAddBottleState {
    bottle: IBottle;
}

@inject("store")
@observer
export class AddBottle extends React.Component<IAddBottleProps, IAddBottleState> {
    private distillery: any;
    private name: any;
    private age: any;
    private descriptionInput: any;

    constructor(props) {
        super(props);

        this.state = {
            bottle: { id: undefined, distillery: "", age: 0, description: "", name: "" }
        };

        this.handleOnAddPress = this.handleOnAddPress.bind(this);
        this.handleOnChange = this.handleOnChange.bind(this);
    }

    public render() {
        const { bottle } = this.state;

        return (
            <div>
                <div style={{ paddingVertical: 20 }}>
                    <div className="">
                        <input
                            ref={(view) => {
                                this.distillery = view;
                            }}
                            value={bottle.distillery}
                            placeholder="Distillery"
                            onChange={this.handleOnChange.bind(this, "distillery")}
                            className="textInput"
                        />
                    </div>
                    <div className="">
                        <input
                            ref={(view) => {
                                this.name = view;
                            }}
                            value={bottle.name}
                            placeholder="Name"
                            onChange={this.handleOnChange.bind(this, "name")}
                            className="textInput"
                        />
                    </div>
                    <div className="">
                        <input
                            ref={(view) => {
                                this.age = view;
                            }}
                            value={bottle.age}
                            placeholder="Age"
                            onChange={this.handleOnChange.bind(this, "age")}
                            className="textInput"
                        />
                    </div>
                </div>
                <div style={{ paddingVertical: 20 }}>
                    <div className="">
                        <textarea
                            placeholder="Description"
                            ref={(view) => { this.descriptionInput = view; }}
                            value={bottle.description}
                            onChange={this.handleOnChange.bind(this, "description")}
                            style={{ minHeight: 60 }}
                            className="textInput"
                        />
                    </div>
                </div>
                <div style={{ paddingVertical: 20 }}>
                    <Button onClick={this.handleOnAddPress} color="#009688">Add Bottle</Button>
                </div>
            </div>
        );
    }

    private handleOnChange = (property, event) => {
        const { bottle } = this.state;
        bottle[property] = event.target.value;

        this.setState({ bottle });
    }

    private handleOnAddPress = () => {
        const { bottleStore, eventStore, navigation, scotchNightStore } = this.props.store;
        const { bottle } = this.state;
        const { currentEvent } = scotchNightStore;

        bottleStore.addBottle(bottle).then((newBottle: IBottle) => {
            eventStore.addBottle(currentEvent, newBottle);

            navigation.goBack();
        });
    }
}

export default AddBottle;