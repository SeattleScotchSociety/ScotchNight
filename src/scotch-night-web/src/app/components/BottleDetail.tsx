import { inject, observer } from "mobx-react";
import * as React from "react";

import { IBottleNote } from "../stores/BottleStore";
import { IRootStore } from "../stores/RootStore";

import { MyNotes } from "./MyNotes";
import { NotesOverview } from "./NotesOverview";

interface IBottleDetailProps { store: IRootStore; }
interface IBottleDetailState {
    notes: IBottleNote;
    resetCount: number;
    view: number;
    rating: number;
}

@inject("store")
@observer
export class BottleDetail extends React.Component<IBottleDetailProps, IBottleDetailState> {
    constructor(props) {
        super(props);

        const { scotchNightStore } = this.props.store;

        this.handleOnPressRating = this.handleOnPressRating.bind(this);
        this.handleSelectView = this.handleSelectView.bind(this);
        this.handleOnResetNotes = this.handleOnResetNotes.bind(this);
        this.handleOnSaveNotes = this.handleOnSaveNotes.bind(this);
        this.handleOnChange = this.handleOnChange.bind(this);

        const notes: IBottleNote = scotchNightStore.memberNotes ? { ...scotchNightStore.memberNotes } : null;

        this.state = {
            resetCount: 0,
            notes,
            rating: 0,
            view: 0
        };
    }

    public componentWillReceiveProps(nextProps) {
        const { scotchNightStore } = this.props.store;

        const notes = scotchNightStore.memberNotes ? { ...scotchNightStore.memberNotes } : null;

        this.setState({
            notes
        });
    }

    public render() {
        const { view, resetCount } = this.state;
        const { currentBottle, summaryNotes, memberNotes } = this.props.store.scotchNightStore;

        if (!currentBottle) {
            return null;
        }

        return (
            <div className="bottle-detail">
                <h1 className="bottle-detail__distillery">{currentBottle.distillery}</h1>
                <h2 className="bottle-detail__name">{currentBottle.name}</h2>
                <div className="btn-group">
                    <button onClick={() => this.handleSelectView(0)} className={`btn ${view === 0 ? "btn--primary" : ""}`}>Overview</button>
                    <button onClick={() => this.handleSelectView(1)} className={`btn ${view === 1 ? "btn--primary" : ""}`}>My Notes</button>
                </div>
                <NotesOverview view={view} notes={summaryNotes} thoughts={memberNotes ? memberNotes.thoughts : ""} tags={memberNotes ? memberNotes.tags : ""} />
                <MyNotes resetCount={resetCount} notes={memberNotes} view={view} onPressRating={this.handleOnPressRating} reset={this.handleOnResetNotes} save={this.handleOnSaveNotes} onChange={this.handleOnChange} />
            </div>
        );
    }

    private handleSelectView(index) {
        this.setState({ view: index });
    }

    private handleOnPressRating(rating: number) {
        this.setState({ notes: { ...this.state.notes, rating } });
    }

    private handleOnChange(noteTitle, value) {
        const update = { ...this.state.notes };
        update[noteTitle] = value;
        this.setState({ notes: update });
    }

    private handleOnSaveNotes() {
        // let { addReview } = this.props.actions;

        // addReview(this.state.notes);

        // this.setState({ view: 0 });
        // // this.view.scrollTo({ x: 0, y: 0, animated: true });
    }

    private handleOnResetNotes() {
        // const notes = Object.assign({}, this.props.navigation.state.params.bottle.memberNotes);
        // this.setState({ resetCount: this.state.resetCount + 1, notes });
    }
}

export default BottleDetail;
