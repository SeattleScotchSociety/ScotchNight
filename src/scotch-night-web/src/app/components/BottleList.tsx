import { inject, observer } from "mobx-react";
import * as React from "react";
import * as _ from 'lodash';

import { IRootStore } from "../stores/RootStore";

interface IBottleListProps { store: IRootStore; }

@inject("store")
@observer
export class BottleList extends React.Component<IBottleListProps> {
    constructor(props: IBottleListProps) {
        super(props);

        this._handleOnClick = this._handleOnClick.bind(this);
    }

    private _handleOnClick(bottle) {
        const { navigation, scotchNightStore } = this.props.store;

        scotchNightStore.setCurrentBottle(bottle);
        navigation.push(`/bottles/${bottle.id}`);
    }

    public render() {
        const { bottles } = this.props.store.bottleStore;
        const bottleElements = _.sortBy(bottles, ['distillery', 'name']).map((bottle) => {
            const { name, distillery } = bottle;

            return (
                <div className="card" key={bottle.id.toString()} onClick={() => this._handleOnClick(bottle)}>
                    <h3 className="card__head">{distillery}</h3>
                    <h4 className="card__subhead">{name}</h4>
                </div>
            );
        });

        return (
            <div className="container">
                <h1 className="margin-bottom-md">Scotch Library</h1>
                {bottleElements}
            </div>
        );
    }
}

export default BottleList;
