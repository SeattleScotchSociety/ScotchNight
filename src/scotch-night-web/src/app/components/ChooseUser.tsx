import { inject, observer } from "mobx-react";
import * as React from "react";
import { Button } from "reactstrap";

import { IMember } from "../stores/MemberStore";
import { IRootStore } from "../stores/RootStore";

interface IChooseUserProps { store: IRootStore; }

@inject("store")
@observer
export class ChooseUser extends React.Component<IChooseUserProps> {
    constructor(props: IChooseUserProps) {
        super(props);

        this.chooseUser = this.chooseUser.bind(this);
    }

    public render() {
        const { members } = this.props.store.memberStore;

        const memberElements = members.map((member) => {
            return (<Button key={member.id} onClick={() => this.chooseUser(member)}>{member.firstName}</Button>);
        });

        return (
            <div>
                <img src="/src/app/images/whiskey-image.jpg" />
                {memberElements}
            </div>
        );
    }

    private chooseUser(member: IMember) {
        const { scotchNightStore } = this.props.store;
        const { push } = this.props.store.navigation;

        scotchNightStore.setCurrentUser(member);

        push("/events");
    }
}

export default ChooseUser;