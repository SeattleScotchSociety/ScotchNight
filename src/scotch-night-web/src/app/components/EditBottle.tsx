import { observer } from "mobx-react";
import * as React from "react";

import { Bottle, IBottle } from "../stores/BottleStore";
import { IRootStore } from "../stores/RootStore";

import { Input } from "./Input";
import { Modal } from "./Modal";

interface IEditBottleProps { store: IRootStore; bottle: IBottle; }
interface IEditBottleState { bottle: IBottle; }

@observer
export class EditBottle extends React.Component<IEditBottleProps, IEditBottleState> {
    private modal: Modal;
    private wrappedInstance: EditBottle;

    constructor(props: IEditBottleProps) {
        super(props);

        const { bottle: editBottle } = props;

        this.state = {
            bottle: { ...editBottle }
        };

        this.handleOnClick = this.handleOnClick.bind(this);
        this.handleOnChange = this.handleOnChange.bind(this);
        this.open = this.open.bind(this);
    }

    private handleOnChange = (property, event) => {
        const { bottle } = this.state;
        bottle[property] = event.target.value;

        this.setState({ bottle });
    }

    private handleOnClick = () => {
        const { bottleStore } = this.props.store;
        const { bottle } = this.state;

        bottleStore.editBottle(bottle).then((editedBottle: IBottle) => {
            this.modal.close();
        });
    }

    public open() {
        this.modal.open();
    }

    public render() {
        const { bottle } = this.state;

        return (
            <Modal className="add-bottle" ref={(modal) => { this.modal = modal; }}>
                <h2 className="margin-bottom-md">Add a New Bottle</h2>
                <Input
                    type="text"
                    value={bottle.distillery}
                    placeholder="Distillery"
                    onChange={this.handleOnChange.bind(this, "distillery")} />
                <Input
                    type="text"
                    value={bottle.name}
                    placeholder="Name"
                    onChange={this.handleOnChange.bind(this, "name")} />
                <Input
                    type="text"
                    value={bottle.age ? bottle.age : ""}
                    placeholder="Age"
                    onChange={this.handleOnChange.bind(this, "age")} />
                <Input
                    type="text"
                    placeholder="Description"
                    value={bottle.description}
                    onChange={this.handleOnChange.bind(this, "description")} />
                <button className="btn btn--primary btn--block" onClick={this.handleOnClick}>Save Bottle</button>
            </Modal>
        );
    }
}

export default EditBottle;
