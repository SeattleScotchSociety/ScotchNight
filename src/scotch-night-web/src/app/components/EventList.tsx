import { inject, observer } from "mobx-react";
import * as React from "react";
import { Button } from "reactstrap";

import { IEvent } from "../stores/EventStore";
import { IRootStore } from "../stores/RootStore";

import * as format from "date-fns/format";

interface IEventListProps { store: IRootStore; }

@inject("store")
@observer
export class EventList extends React.Component<IEventListProps> {
    constructor(props: IEventListProps) {
        super(props);
    }

    private onEventSelected(event: IEvent) {
        const { scotchNightStore, navigation } = this.props.store;

        scotchNightStore.setCurrentEvent(event);

        navigation.push(`/events/${event.id}`);
    }

    public render() {
        const { events } = this.props.store.eventStore;
        const eventElements = events.map((event) => {
            const { date, description, location, title, id } = event;

            return (
                <div className="box" key={id.toString()} onClick={() => this.onEventSelected(event)}>
                    <h3>{format(date, "MMMM D, YYYY")}</h3>
                    <p className="event__title">{title}</p>
                    <p className="event__desc">{description}</p>
                </div>
            );
        });

        return (
            <div className="container">
                <h1 className="page-header">Events</h1>
                <div className="events">
                    {eventElements}
                </div>
            </div>
        );
    }
}

export default EventList;
