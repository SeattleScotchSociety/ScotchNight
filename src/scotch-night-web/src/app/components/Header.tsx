import { observer } from "mobx-react";
import * as React from "react";
import { Link } from "react-router-dom";

import Auth from "../services/Auth";
import { IScotchNightStore } from "../stores/ScotchNightStore";

interface IHeaderProps { scotchNightStore: IScotchNightStore; auth: any; }
interface IHeaderState { menuIsOpen: boolean; }

@observer
export class Header extends React.Component<IHeaderProps, IHeaderState> {
    constructor(props) {
        super(props);

        this.onClickMenu = this.onClickMenu.bind(this);
        this.onClickLink = this.onClickLink.bind(this);

        this.state = {
            menuIsOpen: false
        }
    }

    onClickMenu() {
        const { menuIsOpen } = this.state;
        this.setState({ menuIsOpen: !menuIsOpen });
    }

    onClickLink() {
        this.setState({ menuIsOpen: false });
    }

    render() {
        const { auth } = this.props;
        const { menuIsOpen } = this.state;
        const { currentUser } = this.props.scotchNightStore;

        return (
            <nav>
                <div className="header">
                    <span>Scotch Night</span>
                    { currentUser ?
                        <div className={`hamburger hamburger--squeeze ${menuIsOpen ? 'is-active' : ''}`} onClick={this.onClickMenu}>
                          <div className="hamburger-box">
                            <div className="hamburger-inner"></div>
                          </div>
                        </div> : null }
                </div>
                <ul className={`header-menu ${menuIsOpen ? 'is-active' : ''}`}>
                    <li>
                        <ul>
                            <li><Link to="/events" onClick={this.onClickLink}>Events</Link></li>
                            <li><Link to="/bottles" onClick={this.onClickLink}>Scotch Library</Link></li>
                        </ul>
                    </li>
                    <li><Link to="/login" onClick={(e) => { auth.logout(); }}>Logout</Link></li>
                </ul>
            </nav>
        );
    }
}
