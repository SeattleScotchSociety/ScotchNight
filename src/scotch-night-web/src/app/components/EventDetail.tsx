import { inject, observer } from "mobx-react";
import * as React from "react";

import { IRootStore } from "../stores/RootStore";
import { AddBottle } from "./AddBottle";
import { GenericMap, IMarker } from "./GenericMap";
import { Modal } from "./Modal";
import { TastingMenu } from "./TastingMenu";

import * as format from "date-fns/format";

interface IEventDetailProps { store: IRootStore; }
interface IEventDetailState {
    position: {
        lat: string,
        lng: string
    };
}

@inject("store")
@observer
export class EventDetail extends React.Component<IEventDetailProps, IEventDetailState> {
    private add: AddBottle;

    constructor(props: IEventDetailProps) {
        super(props);

        this.buildEventMap = this.buildEventMap.bind(this);
        this.handleOnAddBottle = this.handleOnAddBottle.bind(this);
    }

    private handleOnAddBottle() {
        const { scotchNightStore } = this.props.store;
        const { navigation } = this.props.store;
        navigation.push(`${scotchNightStore.currentEvent.id}/add-bottle`);
    }

    private buildEventMap(position: any) {
        let center = {
            lat: 47.6062095,
            lng: -122.3320708
        };

        if (position) {
            center = { ...position };
        }

        const markers = [{
            key: "event-marker",
            position: center
        }];

        return (
            <GenericMap
                containerElement={
                    <div style={{ height: `100%` }} />
                }
                mapElement={
                    <div style={{ height: `100%` }} />
                }
                ref={() => null}
                defaultZoom={13}
                defaultCenter={center}
                onClick={() => null}
                markers={markers}
            />
        );
    }

    public render() {
        const { scotchNightStore } = this.props.store;
        const { currentEvent } = scotchNightStore;

        if (!currentEvent) {
            return null;
        }

        const { date, description, location, title } = currentEvent;
        const { position } = location;
        const eventMap = this.buildEventMap(position);

        return (
            <div className="container">
                <h1>{title}</h1>
                <p className="margin-vertical-md"><em>{description}</em></p>
                <p className="event__detail"><i className="fa fa-fw fa-calendar-o" />&nbsp;{format(date, "MMMM D, YYYY")}</p>
                <p className="event__detail"><i className="fa fa-fw fa-clock-o" />&nbsp;{format(date, "h:mm aa")}</p>
                <p className="event__detail"><i className="fa fa-fw fa-map-marker" />&nbsp;{location.name}</p>
                {eventMap}
                <TastingMenu store={this.props.store} />
                <button className="btn btn--primary btn--block" onClick={this.handleOnAddBottle}>Add Bottle</button>
                <AddBottle ref={(add) => { this.add = add; }} store={this.props.store} />
            </div>
        );
    }
}

export default EventDetail;
