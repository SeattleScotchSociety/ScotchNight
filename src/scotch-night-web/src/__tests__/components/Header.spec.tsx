import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";
import React from "react";
import ReactDOM from "react-dom";

import { locations } from "../../../public/locations";
import { Header } from "../../app/components/Header";

import { rootStore } from "../mocks";

configure({ adapter: new Adapter() });

describe("Component: Header", () => {
    it("should render without throwing an error", () => {
        const wrapper = shallow(
            <Header
                scotchNightStore={rootStore.scotchNightStore}
                location={{ pathname: "/" }}
                auth={{ isAuthenticated: () => false }}
            />
        ).toJson;

        expect(toJson(wrapper)).toMatchSnapshot();
    });
});