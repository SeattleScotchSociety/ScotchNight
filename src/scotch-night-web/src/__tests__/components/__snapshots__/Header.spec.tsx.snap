// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Header should render without throwing an error 1`] = `undefined`;
